#166470806 Setup ESLint with HoundCI

coolbeatz71 opened the pull request:
#### What does this PR do?
Setup ESLint with HoundCI 

#### Description of Task to be completed?
- Configure ESLint
- Configure Lint-Stagged
- Integrate HoundCI with badge
#### How should this be manually tested?

N/A

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#166470806](https://www.pivotaltracker.com/story/show/166470806)

#### Screenshots (if appropriate)

N/A

#### Questions:

N/A

Commit:
chore(eslint-houndci): configure eslint and houndci

- create .eslintrc and .eslintignore
- create .hound.yml and add setup rules
- configure prettier

[Delivers #166470806]

Commit:
chore(eslint): setup airbnb style guide

- configure airbnb style guide
- add husky hook to lint before any commit

[Delivers #166470806]

Commit:
chore(houndCI): add houndCI badge

- add HoundCI badge to the README.md

[Delivers #166470806]

Commit:
chore(eslint): configure require-jsdoc to eslint

- ensure jsdoc rules are respected

[Delivers #166470806]

gracelungu commented on .hound.yml:
```diff
@@ -0,0 +1,7 @@
+  ignore_file: .eslintignore
+
+scss:
+  enabled: false
```
@coolbeatz71 Why is Sass disabled?

coolbeatz71 replied:
Ohh yes! @gracelungu It was the default configuration. I need to change it

coolbeatz71 replied:
> Ohh yes! @gracelungu It was the default configuration. I need to change it

Fixed here https://github.com/andela/ah-kgl-wakanda-frontend/pull/5/commits/44f344ad685d3e58cc1ed704aaf2b66d01a2d41e

Commit:
chore(eslint-houndci): Add dummy test to make circle pass

- rebase from develop
- add script for coveralls

[Delivers #166470806]

Commit:
chore(houndci): enable scss linting

- change scss:enabled to be true

[Delivers #166470806]

Commit:
chore(eslint-houndci): make sure travis send coveralls

- add jest.config file to collect coverage

[Delivers #166470806]

Commit:
chore(eslint-houndci): accept change from develop

- solve git conflict

[Delivers #166470806]

Commit:
chore(eslint-houndci): solve maintability and linting errors

- edit and initialState between Header & Home component
- solve houndci issue
- make Header & Home component to be pure function component

[Delivers #166470806]

Commit:
chore(eslint-houndci): solve houndci errors

- solve import/first issues

[Delivers #166470806]

Commit:
chore(eslint): resolve git conflict

- install cypress utilities

[Delivers #166470806]
